packages/app-builder-lib/src/node-module-collector/yarnNodeModulesCollector.ts

```diff
@@ -17,8 +17,10 @@ export class YarnNodeModulesCollector extends NodeModulesCollector {
   removeNonProductionDependencie(tree: DependencyTree) {
     const dependencies = tree.dependencies || {}
     const _dependencies = tree._dependencies || {}
-    if (dependencies && Object.keys(dependencies).length === 0) {
+    if (Object.keys(_dependencies).length > 0 && Object.keys(dependencies).length === 0) {
       tree.dependencies = this.allDependencies.get(`${tree.name}@${tree.version}`)?.dependencies || {}
+      tree.skipCircularDeps = true
```

Comment:
Can we add some logging here? Probably a `debug` level would suffice as we don't know how often this will trigger for users.

```
const { name, version } = tree
tree.dependencies = this.allDependencies.get(`${name}@${version}`)?.dependencies || {}
log.debug({ name, version }, "circular dependency detected")
tree.circularDependencyDetected = true
```

Reply:
I think this also may be a worthwhile time to refactor `yarnNodeModuleCollector` as it's the same logic and duplicated code.
```
export class YarnNodeModulesCollector extends NpmNodeModulesCollector { }
```

Reply:
Very good suggestion. Fixed.